merge branch 8.0

EBII opened the pull request.

EBII committed:
premiere gestion d url

EBII committed:
ajout prepare url

Commit:
On name Change Ok

EBII committed:
pour fork ebii

EBII committed:
compatibilité odoo openerp data

Commit:
essai de test les methodes ne sont pas executees

EBII committed:
update test

EBII committed:
# This is a combination of 5 commits.
# The first commit's message is:
update test

# This is the 2nd commit message:

[FIX] add missing travis

# This is the 3rd commit message:

ajout du test du on_name_change

# This is the 4th commit message:

probleme de duplicité

# This is the 5th commit message:

alerte singleton

EBII committed:
Merge remote-tracking branch 'upstream/8.0' into 8.0

sebastienbeau commented on base_url/LICENSE:
```diff
@@ -0,0 +1,661 @@
+GNU AFFERO GENERAL PUBLIC LICENSE
```
Pas besoin de mettre la licence dans chaque module tu peux la mettre a la racine ça suffit

bealdav replied:
perso, sauvons le cloud et les bb phoques, ceux sont des octets qui réchauffent la planète et n'apportent rien o module.
mais juste perso.

sebastienbeau commented on base_url/README:
```diff
@@ -0,0 +1,24 @@
+# -*- encoding: utf-8 -*-
```
pas besoin de ce fichier

sebastienbeau commented on base_url/__openerp__.py:
```diff
@@ -0,0 +1,51 @@
+# -*- encoding: utf-8 -*-
```
le header n'est pas correct regarde le template de l'oca

sebastienbeau commented on base_url/__openerp__.py:
```diff
@@ -0,0 +1,51 @@
+    'data': ['views/url_view.xml'],
+
+    'demo': [],
+    #les screenshots en v8: mettre dans static/description/, sans déclaration
```
il faut enlever les commentaires

sebastienbeau commented on base_url/model.py:
```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
```
il faut ajouté le header

sebastienbeau commented on base_url/model.py:
```diff
@@ -0,0 +1,119 @@
+    @api.multi
+    def _set_url(self, name=None):
+        """
+        sauvegarder ancienne url si exist et
```
pas de docstring en français

sebastienbeau commented on base_url/model.py:
```diff
@@ -0,0 +1,119 @@
+
+        model_ref = "%s,%s" % (self._name, self.id)
+        _logger.info ("model utilisé : %s ",model_ref)
+        #import pdb; pdb.set_trace()
```
enlevé le code commenté

sebastienbeau commented on base_url/model.py:
```diff
@@ -0,0 +1,119 @@
+        """
+        model_ref = "%s,%s" % (self._name, self.id)
+
+        _logger.info("model referencé : %s ", model_ref)
```
pas de logger en français

sebastienbeau commented on base_url/model.py:
```diff
@@ -0,0 +1,119 @@
+
+
+"""
+tester changement de nom produit : > nouvell url
```
enlevé commenatire en français

sebastienbeau commented on base_url/tests/test_UN.py:
```diff
@@ -0,0 +1,71 @@
+
+
+
+class TestalaCon(unittest.TestCase):
```
il faudra reprendre le nom des tests ;)

EBII committed:
ajout du test du on_name_change

EBII committed:
probleme de duplicité

EBII committed:
alerte singleton

sebastienbeau commented on base_url/model.py:
```diff
@@ -59,24 +61,81 @@ def _set_url(self, name=None):
+            url_key_list =self.env['url.url'].search([('model_id','=',model_ref)]).url_key
+            _logger.info("listes url : %s",url_key_list)
+            x = 0
+            if len(search_key) < 2:
```
Note le cas avec plus de 2 url ne devrait jamais exister, la contraite sql est fait pour ça.

EBII committed:
requirements & base_sale_substate

EBII committed:
docstring in english

Commit:
empty

EBII committed:
tjs empty

Commit:
actualise les url d'un meme modele

EBII committed:
essai 2

EBII committed:
essai 3

EBII committed:
essai 4

EBII committed:
essai flake

EBII committed:
essai flake 2

EBII committed:
essai flake 3

EBII committed:
flake 4

EBII committed:
autopep

EBII committed:
flake 5

EBII committed:
rename method and correct test

EBII committed:
rename method and correct test

EBII committed:
rename Testbaseurl

EBII committed:
test to pass

EBII committed:
test

EBII committed:
test1

EBII committed:
plus de lisibilités comment en fr

EBII committed:
all test ok local comment in en

EBII committed:
essai 3

EBII committed:
essai security add

EBII committed:
essai security add 2

EBII committed:
essai security add 3

EBII committed:
ajout utf8 views

bealdav commented on base_url/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
```
utiliser la version courte indiqué dans les guidelines

bealdav replied:
tu est sur que tu as répondu aux points de la PR ?

bealdav commented on base_url/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+##############################################################################
+
+{
+    'name': 'base_url',
```
Mettre 'Base Url'. C'est le nom du dossier qui compte pour Odoo. Pensons user friendly / market place

bealdav commented on base_url/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+    'url': '',
+    'installable': True,
+    'auto_install': False,
+    'application': False,
```
remove the 2 keys above

bealdav commented on base_url/README.rst:
```diff
@@ -0,0 +1,84 @@
+{module_title}
+==============
+
+This module extends the functionality of ... to support ...
```
rubrique vitale, merci de compléter

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
```
guideline header sur chaque fichier

EBII replied:
seems ok

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+    url_key = fields.Char(string="Url Id", store=True)
+    model_id = fields.Reference(selection='_reference_models',
+                                help="The id of product or category.",
+                                readonly=True,
```
- manque l'arg 'string' ce qui fait qu'on model_id dans l'ui
- le help mérite d'etre plus consistant (1 ligne complète est bienvenue)
- ce n'est pas un model mais 1 record je crois

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+                                help="The id of product or category.",
+                                readonly=True,
+                                )
+    redirect = fields.Boolean('Redirect')
```
missing 'help'

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+                                )
+    redirect = fields.Boolean('Redirect')
+
+    _sql_constraints = [("urlurl unique key",
```
s/urlurl unique key/url unique key   car c un 1 nom

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+
+    @api.model
+    def _reference_models(self):
+        return []
```
s/_reference_models/_referencable_models   ou 1 truc du genre

EBII replied:
no same idea but short

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+
+    _name = "url.url"
+
+    url_key = fields.Char(string="Url Id", store=True)
```
- par défaut il est déjà True un field Char, à enlever
- s/Url Id/Url

bealdav commented on product_url/models/model.py:
```diff
@@ -0,0 +1,31 @@
+        res = super(UrlUrl, self)._reference_models()
+
+        res += [("product.template", "Product"),
+                ('product.category', "Category")]
```
Perso j'utilise ' pour les infos courtes
    et " pour les infos longues.
Tout le monde ne fait pas comme ça, mais tout le monde est constant, fix it

bealdav commented on product_url/demo/url.url.csv:
```diff
@@ -0,0 +1,7 @@
+2,"product.template,2",manutoring,True
+3,"product.template,2",monitoring so good,True
+4,"product.template,2",monitoring betteur,True
+5,"product.template,2",monitoring bourré,False
```
english humor please

bealdav commented on product_url/__openerp__.py:
```diff
@@ -0,0 +1,53 @@
+    # v8: ./static/description/icon.png
+    # v9: ./static/description/icon.png
+
+    'url': '',  # ??
```
Used key by Odoo

bealdav commented on product_url/__openerp__.py:
```diff
@@ -0,0 +1,53 @@
+    # rien à mettre dans __openerp__.py
+    # v7: ./static/src/img/icon.png
+    # v8: ./static/description/icon.png
+    # v9: ./static/description/icon.png
```
remove comment

bealdav commented on base_url/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+    # any module necessary for this one to work correctly
+    'depends': [],
+    'external_dependencies': {
+        'bin': [],
```
not used, remove

bealdav replied:
car je suis pas sur que soit allé jusqu'au bout

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+    url_key = fields.Char(compute="_compute_url",
+                          inverse="_inverse_set_url", string="Url key")
+    redirect_url_key_ids = fields.One2many(compute="_compute_redirect_url",
+                                           comodel_name="url.url")
```
- s / redirect_url_key_ids / url_redirect_key_ids
- missing 'inverse_name' arg

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+    _name = "abstract.url"
+
+    url_key = fields.Char(compute="_compute_url",
+                          inverse="_inverse_set_url", string="Url key")
```
- s / Url key / Url Key
- missing 'help' arg

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
```
s / model.py / url.py

bealdav replied:
le but est que le reviewer perde le moins de temps possible il faut finir le boulot à 105 %

bealdav commented on base_url/tests/test_un.py:
```diff
@@ -0,0 +1,18 @@
+
+    def test_upper(self):
+        # import pdb; pdb.set_trace()
+        self.assertEqual('foo'.upper(), 'FOO')
```
Why these tests ?

bealdav commented on base_url/views/url_view.xml:
```diff
@@ -0,0 +1,43 @@
+<data>
+
+    <record id="url_view_form" model="ir.ui.view">
+
```
saut de ligne inutile

bealdav commented on base_url/views/url_view.xml:
```diff
@@ -0,0 +1,43 @@
+    </record>
+
+        <record id="url_view_tree" model="ir.ui.view">
+
```
<record pas aligné avec précédent

bealdav commented on base_url/views/url_view.xml:
```diff
@@ -0,0 +1,43 @@
+        </record>
+
+        <record id="base_url_action_view" model="ir.actions.act_window">
+        <field name="name">url list Viewer</field>
```
pour viewer ? on peut pas modifier ?

bealdav commented on base_url/views/url_view.xml:
```diff
@@ -0,0 +1,43 @@
+            </field>
+        </record>
+
+        <record id="base_url_action_view" model="ir.actions.act_window">
```
s / base_url_action_view / base_url_action     car c 1 action pas une vue

bealdav commented on product_url/README.rst:
```diff
@@ -0,0 +1,85 @@
+Installation
+============
+
+To install this module, you need to:
```
manque description

EBII replied:
si ligne 9

bealdav replied:
cela ne me dit rien sur ce que fait le module. Encore moins à ceux qui le verront sur l'app store.

bealdav commented on product_url/__openerp__.py:
```diff
@@ -0,0 +1,53 @@
+    'version': '8.0.0.1.0',
+    'category': 'tools',
+    'license': 'AGPL-3',
+    'summary': 'keep history of url for product & categorie  ',
```
s / for product & categorie / for products & categories

bealdav commented on product_url/__openerp__.py:
```diff
@@ -0,0 +1,53 @@
+    'summary': 'keep history of url for product & categorie  ',
+
+    'author': 'Akretion',
+    'website': 'https://www.akretion.fr',
```
http://www.akretion.com/fr

bealdav commented on product_url/models/model.py:
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
```
le dossier s'appel model, le fichier pourrait s'appeler url ou product

bealdav commented on product_url/tests/test_two.py:
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+import unittest
+
```
utiliser TransactionCase qui est une surcharge de unittest

bealdav commented on product_url/tests/test_two.py:
```diff
@@ -0,0 +1,22 @@
+        with self.assertRaises(TypeError):
+            s.split(2)
+
+if __name__ == '__main__':
```
pas utile, le test est appelé quand meme pas le framework odoo

bealdav commented on product_url/tests/test_two.py:
```diff
@@ -0,0 +1,22 @@
+        self.assertEqual(s.split(), ['hello', 'world'])
+        # check that s.split fails when the separator is not a string
+        with self.assertRaises(TypeError):
+            s.split(2)
```
test pas trop utile pour ton use case

bealdav commented on product_url/tests/test_un.py:
```diff
@@ -0,0 +1,18 @@
+
+    def test_upper(self):
+        # import pdb; pdb.set_trace()
+        self.assertEqual('foo'.upper(), 'FOO')
```
pas trop utile

EBII replied:
a mieux voir passer le test,

bealdav replied:
pas compatible OCA, dsl

bealdav commented on product_url/views/product_view.xml:
```diff
@@ -0,0 +1,27 @@
+                        <page string="Rewrite url">
+                            <group>
+                                 <field name="url_key"/>
+                                    <field name="redirect_url_key_ids"/>
```
affiche la vue par défaut du o2m.
Là on ne veut pas la colonne model_id (mettre à invisible) car il ne fait que répéter le nom du produit

bealdav commented on product_url/views/product_view.xml:
```diff
@@ -0,0 +1,27 @@
+            <field name="model">product.template</field>
+            <field name="inherit_id" ref="product.product_template_form_view"/>
+            <field name="arch" type="xml">
+
```
mise en page améliorable

bealdav commented on product_url/tests/test_url.py:
```diff
@@ -0,0 +1,68 @@
+        super(Testbaseurl, self).setup()
+
+    def test_change_name(self):
+        product = self.env['product.template'].browse(9)
```
pas d'id pour les produits, trouver l'xml id

bealdav commented on product_url/views/product_view.xml:
```diff
@@ -0,0 +1,27 @@
+                <!--<xpath expr="//field[@name='list_price']" position="after">  </xpath>-->
+                    <!-- Add your fields or attributes here -->
+                    <notebook>
+                        <page string="Rewrite url">
```
quand tu as beaucoup d'onglet ça rentre plus
je suggère Url

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+        return []
+
+    @api.multi
+    def _get_object(self, urls):
```
urls est un itérable (list, etc) ?
- si oui ("url_key", "in", urls)
- si non s / urls / url

EBII replied:
ok url

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+        """
+        :return: return object attach to the url
+        """
+        object = self.search([("url_key", "=", urls)]).model_id
```
que se passe-t-il la recherche vaut False ?

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+        :return: return object attach to the url
+        """
+        object = self.search([("url_key", "=", urls)]).model_id
+        return object
```
on pourrait faire directment return self.search([("url_key", "=", urls)]).model_id

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+
+    def _prepare_url(self, name=None):
+        url_key = "prepare"
+        if name is None:
```
inverser la condition permet de mettre : 
`if name:`  au lieur de  `if name is None:`

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+        url_key = "prepare"
+        if name is None:
+            url_to_normalize = self.url_key
+            url_key = slugify(url_to_normalize)
```
pourquoi une var intermédiaire ?

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+            return {'value': {},
+                    'warning': {
+                        'title': 'Adapt text rules',
+                        'message': 'it will will be adapted to %s' % (url)}}
```
au niveau usability, je pense que c 1 peu lourd. Je ferais pas de warning ; voir avec seb

EBII replied:
no it is hope that you write tùttù and it was change so we choose to prompt the user about the change ..

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+
+    @api.onchange('url_key')
+    def on_url_key_change(self):
+        url = self._prepare_url(self.url_key)
```
en theorie c mieux avec for record in self:
même si l'editeur ne le fait pas mais tu l'as fait dans la fct précedente

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+        _logger.info("reference model : %s ", model_ref)
+
+        self.redirect_url_key_ids = self.env["url.url"]\
+            .search([('model_id', '=', model_ref), ('redirect', '=', True)])
```
sans doute un gout personnel, mais trop de saut de ligne nuit à la lisibilité de l'ensemble du script
=> 2 lignes vides pas vraiment utiles ?

bealdav commented on base_url/models/model.py:
```diff
@@ -0,0 +1,143 @@
+    def _compute_redirect_url(self):
+        """
+        :return:
+
```
?

EBII committed:
correct

EBII committed:
gitignore

EBII committed:
re-essai

EBII committed:
re-essai1

EBII committed:
re-essai2

EBII committed:
re-essai3

EBII committed:
oh squash

docstring in english

empty

tjs empty

actualise les url d'un meme modele

 essai 2

 essai 3

 essai 4

essai flake

essai flake 2

essai flake 3

flake 4

autopep

flake 5

rename method and correct test

rename method and correct test

rename Testbaseurl

test to pass

 test

 test1

plus de lisibilités comment en fr

all test ok local comment in en

essai 3

essai security add

essai security add 2

essai security add 3

ajout utf8 views

correct

gitignore

re-essai

re-essai1

re-essai2

re-essai3

EBII committed:
Merge branch '8.0' of github.com:EBII/ak-web into 8.0
squash is alive

Commit:
ajout category

EBII committed:
correction pb create on_change

Commit:
vue cat

EBII committed:
pylint

EBII committed:
pylint

EBII committed:
openerp__

EBII committed:
header

EBII committed:
header

EBII committed:
pr...

EBII committed:
pr...urls/url

EBII committed:
notebook name

bealdav commented on product_url/__openerp__.py:
```diff
@@ -0,0 +1,32 @@
+#    @author EBII MonsieurB <monsieurb@saaslys.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+#
+##############################################################################
```
A 3 lignes près (à supprimer) on y est :-)

sebastienbeau replied:
regarde dans le template de l'oca, l'année est fausse ;)

bealdav commented on product_url/README:
```diff
@@ -0,0 +1,24 @@
+# -*- encoding: utf-8 -*-
```
fichier inutile

EBII committed:
review code

EBII committed:
Merge branch '8.0' of github.com:EBII/ak-web into 8.0
interrogation ?

EBII committed:
reset code

EBII committed:
Merge branch '8.0' of github.com:EBII/ak-web into 8.0
merge reset code moins de commit

EBII committed:
README.rst

EBII committed:
module description

EBII committed:
no test for test test

EBII committed:
reformat code xml

EBII committed:
ajout test on_change_url_key

EBII committed:
supression de la class moc WarningTestMixin du test

EBII committed:
test onchange_catagory_url_key

sebastienbeau commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+.idea*
+*.pyc
+*~
+Brouillons.txt
```
bizarre de mettre cette ligne la, a mon avis il serait mieux de la mettre dans ton gitignore perso (qui est global en plus)

EBII committed:
test onchange_category_url_key

EBII committed:
Merge branch '8.0' of github.com:EBII/ak-web into 8.0
youpi j efface des commits

sebastienbeau commented on base_url/models/base_url.py:
```diff
@@ -0,0 +1,146 @@
+                    'redirect': False}
+            self.env['url.url'].create(vals)
+            # other url of object set redirect to True
+        this_model_urls = self.env['url.url'].search([
```
il faudrait mettre un nom de variable plus explicite
genre redirect_urls = ...

tu peux éviter du temps de traitement inutile en ajoutant ('redirect', '=', False) dans le filtre car ceux qui sont déjà a true ça sert a rien de les retraiter

sebastienbeau commented on base_url/models/base_url.py:
```diff
@@ -0,0 +1,146 @@
+
+        if already_exist_url:
+            # existing key in wich object ?
+            for model in already_exist_url.model_id:
```
le for n'est pas utile ici car
1 - si jamais il y avait plusieurs élément suite a la recherche alors la liste serait already_exist_url donc le already_exist_url.model_id remontrait un message d'erreur. il aurait plutôt fallut écrire
for url in already_exist_url: 
    url.model_id.....
....
2- ce cas n'est pas possible car il y a une contrainte sql qui empeche d'avoir plus de 2 urls

sebastienbeau commented on base_url/models/base_url.py:
```diff
@@ -0,0 +1,146 @@
+    @api.multi
+    def _compute_url(self):
+        for record in self:
+            model_ref = "%s,%s" % (record._name, record.id)
```
ce code est répété a plusieurs endroits. Malheuresement il n'y a pas de méthode magique dans odoo. Par contre pour faire plus propre ça peut etre mieux de faire une méthode dans AbstractUrl genre 

def _get_reference(self):
    return "%s,%s" % (self.name, self.id)

comme ça le jour ou odoo change pour une "," à un ";" et bien c'est plus clean a mettre a jour

EBII replied:
ok sauf que c'est bien _name et pas name la valeur souhaité...;o)

EBII committed:
test onchange_category_url_key

sebastienbeau commented on base_url/models/base_url.py:
```diff
@@ -0,0 +1,146 @@
+            url = record.env["url.url"].search([('model_id', '=', model_ref),
+                                                ('redirect', '=', False)])
+            if url:
+                record.url_key = url[0].url_key
```
le url[0] ne sert a rien car tu retourne tjs qu'un seul élement (voir contrainte sql). Il faut mieux mettre un url tout court car si il y en a plus c'est qu'on a un bug et donc ça plantera au lieu de rien dire

EBII replied:
attention les category passe en listes je vais voir

sebastienbeau commented on base_url/models/base_url.py:
```diff
@@ -0,0 +1,146 @@
+                                           comodel_name='url.url')
+
+    def _prepare_url(self, name=None):
+        url_key = 'prepare'
```
a mon avis cette variable est un truc de test

sebastienbeau commented on base_url/models/base_url.py:
```diff
@@ -0,0 +1,146 @@
+
+        for record in self:
+            if record.url_key:
+                url = record._prepare_url(record.url_key)
```
je pense pas qu'il faille utilise la meme méthode prepare_url, car en regardant la méthode tu fini par mettre un if pour traiter différement le cas des name et des urlkey.

A mon avis il faut utiliser la méthode prepare uniquement dans le cas du onchange sur les noms.

Dans le cas du onchange sur les url tu peux simplement faire ça
new_url = slugify(record.url_key)
if new_url != record.url_key:
   record.url_key = new_url
   return {...warning...}

De plus la méthode prepare ne doit pas prendre d'argument en parametre, on a déjà accès a l'object avec toute les valeurs

sebastienbeau commented on base_url/models/base_url.py:
```diff
@@ -0,0 +1,146 @@
+                                'title': 'Adapt text rules',
+                                'message': 'it will will be adapted to %s' %
+                                           (url)}}
+                record.url_key = url
```
ligne non utile car si l'url est identique ça sert a rien de réengistrer la valeur en db

sebastienbeau commented on product_url/README.rst:
```diff
@@ -0,0 +1,86 @@
+Contributors
+------------
+
+* Firstname Lastname <email.address@example.org>
```
pense a mettre ton nom + mail

sebastienbeau commented on product_url/models/product_url.py:
```diff
@@ -0,0 +1,35 @@
+    @api.model
+    def _reference_models(self):
+        res = super(UrlUrl, self)._reference_models()
+
```
saut de ligne pas utile

sebastienbeau commented:
Des petites remarques mais ça avance, ça sera un joli module

EBII committed:
factorisation model ref et correctif ajout test methode _get_model_id_reference

EBII committed:
factorisation model ref et correctif ajout test methode _get_model_id_reference

bealdav commented on product_url/views/category_view.xml:
```diff
@@ -0,0 +1,20 @@
+                            <group string="Rewrite url">
+                                 <field name="url_key"/>
+                                    <field name="redirect_url_key_ids"/>
+                            </group>
```
Reformate ce fichier, stp

bealdav commented on product_url/__openerp__.py:
```diff
@@ -0,0 +1,32 @@
+#    @author EBII MonsieurB <monsieurb@saaslys.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+#
+##############################################################################
```
ligne 2, 3, 8 et 9 à supprimer.
Pourquoi différer du template de base ?

bealdav commented on base_url/__openerp__.py:
```diff
@@ -0,0 +1,32 @@
+#    @author EBII MonsieurB <monsieurb@saaslys.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+#
+##############################################################################
```
some lines to remove, see below

bealdav commented on base_url/README.rst:
```diff
@@ -0,0 +1,84 @@
+#. (root) pip install slugify
+
+Configuration
+=============
```
If no configuration, remove useless blocks

bealdav commented on base_url/README.rst:
```diff
@@ -0,0 +1,84 @@
+
+==============
+{Base URL}
+==============
```
why { ?

bealdav commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+.idea*
+*.pyc
+*~
```
missing break line

EBII replied:
ok

bealdav commented on product_url/tests/__init__.py:
```diff
@@ -0,0 +1 @@
+from . import test_product, test_category
```
split into 2 lines (OCA recommandation for better traceability when changes)

bealdav commented:
Some minor remarks here. Thanks to fix it. After OK for me

EBII committed:
code reviews

EBII committed:
prepare url sans argument

EBII committed:
prepare url sans argument

EBII committed:
test getters url redirect_urls

EBII committed:
test reference model

EBII committed:
test get reference model

EBII committed:
test selection reference correctif

EBII committed:
test selection reference correctif

EBII committed:
AH PAS VU
